WPFHospitalEditor/SearchResultDialog.xaml.cs

```diff
@@ -138,6 +151,48 @@ private void addAdvancedSearchButton()
             Grid.SetColumn(advancedSearch, 4);
 
             DynamicGrid.Children.Add(advancedSearch);
+
+            advancedSearch.Click += (s, e) =>
+            {
+                if (row.ContainsKey(Grid.GetRow(advancedSearch)))
+                {
+                    row[Grid.GetRow(advancedSearch)].selected = true;
+                    mapObjectController.update(row[Grid.GetRow(advancedSearch)]);
+                    if (row[Grid.GetRow(advancedSearch)].Description.Equals(""))
+                    {
+                        CanvasService.addObjectToCanvas(mapObjectController.getOutterMapObjects(), HospitalMap.canvasHospitalMap);
+                        this.Close();
+                    }
+                    else
+                    {
+                        string[] split = getBuildingAndFloor(row[Grid.GetRow(advancedSearch)]).Split("-");
+                        List<MapObject> chosenBuilding = findBuilding(split[0]);
+                        Building buildingFromSearch = new Building(chosenBuilding, int.Parse(split[1]), role);
+                        Building.canvasBuilding.Children.Clear();
+                        CanvasService.addObjectToCanvas(getObjects(split[0], split[1]), Building.canvasBuilding);
+                        buildingFromSearch.Owner = hospitalMap;
+                        buildingFromSearch.Show();
+                        hospitalMap.Hide();
+                        this.Close();
+                    }
+                }
+            };
```

Comment:
Can this be extracted in separate method?

Reply:
Yes, it's separated now.